Structure.h

```diff
@@ -0,0 +1,215 @@
+//
+// Created by dobrikov on 9/24/2019.
+//
+
+#ifndef SPISOK_2_STRUCTURE_H
+#define SPISOK_2_STRUCTURE_H
+
+#include <iostream>
+#include "my_exceptions.h"
+
+myexception ex2;
+
+class Structure {
+protected:
+    struct List
+    {
+        int num;
+        List* next;
+        List* prev;
+        List(): next(NULL), prev(NULL){};
+    }*begin, *end;
+    int size = 0;
+public:
+    Structure(): begin(NULL), end(NULL){};     //constructor
+    ~Structure(){};   //destructor
+    virtual void push(int n) {};       //base methods
+    virtual void push_front(int n) {};
+    virtual void pop() {};
+    virtual void pop_back() {};
+    virtual int head_element() {return 0;};
+    virtual int end_element() { return 0;};
+    virtual int print_size() {return 0;};
+};
+
+class myQueue: public Structure{
+public:
+    myQueue() {};
+    ~myQueue()
+    {
+        while(begin)
+        {
+            List* tmp = begin;
+            begin = begin->next;
+            delete tmp;
+        }
+    }
+
+    void push(int n)
+    {
+        if(end != NULL)
+        {
+            List* temp = new List;
+            temp->num = n;
+            temp->next = NULL;
+            end->next = temp;
+            end = end->next;
+        }
+        else
+        {
+            end = new List;
+            end->num = n;
+            end->next = NULL;
+            begin = end;
+        }
+        size++;
+    }
+
+    void pop()
+    {
+        try{
+            if(begin == NULL) {throw ex2;}
+            if (begin == end) end = NULL;
+            List* el = begin;
+            begin = begin->next;
+            delete el;
+            size--;
+        }
+        catch(std::exception& e){
+            std::cout << e.what() << std::endl;
+        }
+    }
+
+    int head_element()
+    {
+        if(begin != NULL)
+        {
+            int el = begin->num;
+            return el;
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    int print_size()
+    {
+        return size;
+    }
+
+    void Print()
+    {
+        try{
+            List* temp = begin;
+            if(temp == NULL)
+            {
+                throw ex2;
+            }
+            while (temp != NULL)
+            {
+                int el = temp->num;
+                std::cout << el << " ";
+                temp = temp->next;
+            }
+            std::cout << std::endl;
+        }
+        catch (std::exception& e){
+            std::cout << e.what() << std::endl;
+        }
+
+    }
+};
+
+class myDequeue: public myQueue {
```

Comment:
Same. Move Deque class to separate file. Also pay attention to naming: structure is named Deque, not Dequeue.